internal/voting/sql.go

```diff
@@ -51,6 +51,16 @@ const (
 		id BYTEA PRIMARY KEY
 	);`
 
+	tableHeight = `CREATE TABLE IF NOT EXISTS ` + votingSchemaName + `.height (
+		name TEXT PRIMARY KEY, -- name is 'height'
+		height INT NOT NULL
+	);`
+
+	getHeight = `SELECT height FROM ` + votingSchemaName + `.height WHERE name = 'height';`
```

Comment:
It seems weird to put the height in the voting schema, right? It is never actually used in relation to the rest of the votes / resolutions.

Reply:
In the sense that it's not referenced by other tables, yes.  But it is convenient from the perspective of building the app and not having to create another store `kwild_internal` was really supposed to be for use internal to the `pg` package, although engine uses it too now for schemas.

The most "blockchainy" store seemed to be `voting`, so it's not terrible.  What's a better spot?  Not opposed to moving it, just lacking a better suggestion.

Reply:
I don't have other suggestions other than creating another schema, however it does seem like we are redundantly storing this height.

In ABCI, we store both the block height and app hash in a local badger DB. ABCI uses this to know where the app left off on startup (called on (*AbciApp).Info()`. It seems like we are potentially duplicating this functionality in part here.

IMO, it seems like TxApp should separately store both the height and apphash, and return this to info. CometBFT will actually catch these discrepancies automatically. Tracking chain data for synchronicity purposes seems like it should be a duty of TxApp, and so I feel like chain metadata should be a different store?

Reply:
Sort've related, but in the future it may be worthwhile to group all stores in a sub package of internal. Logically grouping the functions in their own packages is nice, but it does sort've bloat the top-level `internal`

Reply:
I just spent a little bit of time testing this out. For the sake of this release, I think leaving height in vote store is ideal.

Reply:
The motivation for also putting height (or block hash ideally) in postgres is to ensure we can't accidentally start up with an old or fresh postgres database and new blockchain data (in `root-dir`) without realizing it until much later, like when a tx fails.  https://github.com/kwilteam/kwil-db/pull/565#issuecomment-1973459908  So the redundancy is for consistency between the two.

Both the badger DB and cometbft's blockchain data is in `root-dir`, and we're trying to tackle cases where your `root-dir` has data that is different from postgres.

Reply:
Comet relies on badger db to ensure consistency though. But now that we are using postgres, this seems incorrect; we want comet to be checking against postgres. We could check both, but IMO it seems like the badger db is almost useless

Reply:
Oh, yeah, I agree with that.